test/common/index.js

```diff
@@ -51,6 +51,7 @@ const hasCrypto = Boolean(process.versions.openssl);
 // If the binary was built without-ssl then the crypto flags are
 // invalid (bad option). The test itself should handle this case.
 if (process.argv.length === 2 &&
+    !process.env.NODE_SKIP_FLAG_CHECK &&
```

Comment:
```suggestion
    !('NODE_SKIP_FLAG_CHECK' in process.env) &&
```

That would include empty values but it's just my personal preference (without looking it up: process.env should have a null prototype).

Reply:
I think that would be unusual usage of environment variables – typically, programs check whether the value is empty or not, not whether the variable is technically set or not. As far as I know, this is because in shell scripting, the two conditions become hard to distinguish anyway.

> process.env should have a null prototype

It does not, unfortunately.

Reply:
> It does not, unfortunately.

Do you know if there's a reason for that? I guess we should otherwise change that.

> typically, programs check whether the value is empty or not, not whether the variable is technically set or not. As far as I know, this is because in shell scripting

:+1:

Reply:
> I guess we should otherwise change that.

I assume it’s backwards compatibility. I’d personally be careful about changing it, because `process.env.hasOwnProperty()` seems to be something that might be used a lot, and the advantages from changing might not be all that big. (For example, I wouldn’t consider it a security issue that environment variables might override `Object.prototype` entries.)

Reply:
Your hunch is correct (I just checked `gzemnid` and there are indeed quite a few modules using that).